cruisetrack/process/workflow_lines.py

```diff
@@ -73,28 +59,72 @@ def process_lines(layer_provider, laye_r, is_individual_trackline, is_accessory,
-            for line in geom:
-                start_point = QgsPointXY(geom[0][0])
-                end_point = QgsPointXY(geom[-1][-1])
-                fea_t["X_start"] = start_point[0]
-                fea_t["X_stop"] = end_point[0]
-                fea_t["Y_start"] = start_point[1]
-                fea_t["Y_stop"] = end_point[1]
-                # ellipsoid; noch nicht herausgefunden, wie man das umstellt (trotz geänderter Projektion)
-                fea_t["length"] = fea_t.geometry().length()
-                if (abs(start_point.azimuth(end_point)) > 45) and (abs(start_point.azimuth(end_point)) < 125):
-                    fea_t["X_mean"] = median([start_point[1], end_point[1]])
-                else:
-                    fea_t["X_mean"] = median([start_point[0], end_point[0]])
-                laye_r.updateFeature(fea_t)
-        laye_r.commitChanges()
-        df = pd.DataFrame(fea_t.attributes() for fea_t in laye_r.getFeatures(QgsFeatureRequest()))
+            geom = fea_t.geometry().constGet()
+            for n in list(range(len(geom[0]))):
+                vertices_on_line = QgsPointXY(geom[0][n])
+                lon_series[coun_t, n] = vertices_on_line[0]
+                lat_series[coun_t, n] = vertices_on_line[1]
+            coun_t = coun_t + 1
+
+        lon = []
+        lat = []
+
+        for n in list(range(coun_t)):
+            lon = lon + list(filter(lambda num: num != 0, lon_series[n]))
+            lat = lat + list(filter(lambda num: num != 0, lat_series[n]))
+
+        arrays = [lon, lat]
+        df = pd.DataFrame(arrays)
+        df = df.T
+    df = df.rename(columns={0: 'x', 1: 'y'})
+    return df
+
+
+def lines_workflow(layer_provider, laye_r, is_individual_trackline, is_accessory, is_nonebt,
+                   is_normal_profile, flip_we, only_process_2nds, is_littorina, flip_ns):
+    field_name = "X_start"  # if there are the right fields still missing, make them
+    field_index = laye_r.fields().indexFromName(field_name)
+    if field_index == -1:
+        layer_provider.addAttributes([QgsField("X_start", QVariant.Double),
+                                      QgsField("X_stop", QVariant.Double), QgsField("Y_start", QVariant.Double),
+                                      QgsField("Y_stop", QVariant.Double), QgsField("length", QVariant.Double),
+                                      QgsField("X_mean", QVariant.Double)])
+        laye_r.updateFields()
+
+    # put data from shapefile into attribute table
+    coun_t = laye_r.featureCount()
+
+    """ if just one line, with multiple points, 
+    but as normal regular profile lines wanted (additinoally accessory tasks)"""
+    single_line = True if coun_t == 0 else False
+    df = layer_to_dataframe(laye_r=laye_r,
+                            single_line=single_line,
+                            is_individual_trackline=is_individual_trackline)
+    # df.to_csv('/home/markus/scripting/cruise_track/tests/data/line_layer_as_df.csv', index=False)
```

Comment:
This is the pickle part, but it is currently deactivated.

We need to make more test_cases.